### PR TITLE
Issue #707 - Transient is not needed anymore

### PR DESCRIPTION
**Pull request by dalifreire**
Issue #707. Transient requirement has been removed from qulice and is not needed anymore.
This PR do the following changes:

- Removes transient everywhere where it was not added explicitly.
- Removes 'of' parameter from all '@EqualsAndHashCode'.
- Fixes some junit tests.

### CODE REVIEW

**Comment by dalifreire** [comment]
@davvd Please, find someone to review.

**Comment by davvd** [comment]
@dalifreire Let me find a reviewer for this pull request, thanks for submitting it

**Comment by davvd** [comment]
@mkordas it's yours,please review

**Comment by mkordas** [comment]
@dalifreire I'll check it today

**Comment by mkordas** [comment]
@dalifreire I'm on it now

File: src/main/java/org/takes/facets/auth/social/PsFacebook.java [hunk 1]
```diff
@@ -91,22 +91,22 @@
...
      * Request for fetching app token.
      */
-    private final transient com.jcabi.http.Request request;
+    private final com.jcabi.http.Request request;
 
     /**
      * Facebook login request handler.
      */
-    private final transient WebRequestor requestor;
+    private final WebRequestor requestor;
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/social/PsFacebook.java]: @dalifreire are you sure that these new fields can go to equals? Do we have any tests for it?

File: src/main/java/org/takes/facets/auth/social/PsGithub.java [hunk 2]
```diff
@@ -75,22 +75,22 @@
...
      * Key.
      */
-    private final transient String key;
+    private final String key;
 
     /**
      * GitHub OAuth url.
      */
-    private final transient String github;
+    private final String github;
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/social/PsGithub.java]: @dalifreire same here, we extend `equals` here

File: src/main/java/org/takes/facets/auth/social/PsLinkedin.java [hunk 3]
```diff
@@ -51,7 +51,7 @@
  * @version $Id$
  * @since 0.11.3
  */
-@EqualsAndHashCode(of = { "app", "key" })
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/social/PsLinkedin.java]: @dalifreire do you think it was intentional to not put all fields here?

File: src/main/java/org/takes/facets/auth/social/PsGoogle.java [hunk 4]
```diff
@@ -88,27 +88,27 @@
...
      * Redirect URI.
      */
-    private final transient String redir;
+    private final String redir;
 
     /**
      * Google OAuth url.
      */
-    private final transient String gauth;
+    private final String gauth;
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/social/PsGoogle.java]: @dalifreire should we really use auth URLs in our hashCode?

File: src/main/java/org/takes/facets/auth/social/PsTwitter.java [hunk 5]
```diff
@@ -74,22 +74,22 @@
...
      * Key.
      */
-    private final transient String key;
+    private final String key;
 
     /**
      * Request for fetching app token.
      */
-    private final transient com.jcabi.http.Request token;
+    private final com.jcabi.http.Request token;
```
Comment (mkordas) [inline src/main/java/org/takes/facets/auth/social/PsTwitter.java]: @dalifreire I think it was fne that some token didn't belong to an `equals`

**Comment by mkordas** [comment]
@dalifreire see my comments above

**Comment by mkordas** [comment]
@dalifreire ping

**Comment by dalifreire** [comment]
@mkordas Thanks for your review. Could you check again?

**Comment by mkordas** [comment]
@dalifreire sure, I'll check it today

**Comment by mkordas** [comment]
@dalifreire I'm on it

**Comment by mkordas** [comment]
@dalifreire looks good, thanks!

**Comment by mkordas** [comment]
@rultor merge

**Comment by mkordas** [comment]
@yegor256 ping here

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@ypshenychka please, let us know what do you think about this ticket, according to our [QA rules](http://at.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by ypshenychka** [comment]
@davvd Quality is good.

**Comment by davvd** [comment]
> @davvd Quality is good.

@ypshenychka thank you

**Comment by davvd** [comment]
@mkordas paied 10 mins to @ypshenychka for QA review (payment ID is `97964897`)

thank you, added **29 mins** to your acc, payment num is `57bd9f542cdf8258b7041555`, 117 hours and 16 mins spent in total to complete

review comments (c=14) added as a bonus

+29 added to your rating, current score is: [+4808](http://www.netbout.com/b/36080?open=rating)

**Pull request merged**